### PR TITLE
fix: resolve PVC allocation missing warnings in ETL process

### LINKED ISSUE #3211: Missing allocations for PVC asset keys
**Describe the bug**
Opencost reports: WRN ETL: did not find allocations for asset key: default-cluster/pvc-01c2ff8c-b3d5-4a01-86f0-9372d9244e00

**To Reproduce**
1. kubectl -n opencost logs opencost-xxxxxxxxx | grep 01c2ff8c-b3d5-4a01-86f0-9372d9244e00

However this PVC is bound:

apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    pv.kubernetes.io/bind-completed: "yes"
    pv.kubernetes.io/bound-by-controller: "yes"
    volume.beta.kubernetes.io/storage-provisioner: ebs.csi.aws.com
    volume.kubernetes.io/selected-node: ip-10-208-56-204.eu-west-1.compute.internal
    volume.kubernetes.io/storage-provisioner: ebs.csi.aws.com
  creationTimestamp: "2024-10-11T10:25:43Z"
  finalizers:
  - kubernetes.io/pvc-protection
  labels:
    opster.io/opensearch-cluster: tst-opensearch
    opster.io/opensearch-nodepool: masters
  name: data-tst-opensearch-masters-2
  namespace: default
  resourceVersion: "892677347"
  uid: 01c2ff8c-b3d5-4a01-86f0-9372d9244e00
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 10Gi
  storageClassName: gp3
  volumeMode: Filesystem
  volumeName: pvc-01c2ff8c-b3d5-4a01-86f0-9372d9244e00
status:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 10Gi
  phase: Bound

**Expected behavior**
I've enabled CloudCost. OpenCost is querying our AWS Athena database successfully 

**Which version of OpenCost are you using?**
1.114.0 (dcbd21f)

**Additional context**
There are multiple did not find allocations for PVC asset keys, otherwise opencost works fine

### PR DESCRIPTION
**Pull request by fromsaurav**
## Summary
Fixes false warnings for bound PVCs:  
`ETL: did not find allocations for asset key: default/pvc-xxx`

## Root Cause
PVC asset keys never matched allocation keys because:
- **PVC assets:** `cluster/pvc-uuid` (cluster-level resources)  
- **Allocations:** `cluster/node` (node-specific)  
- PV costs were already correctly allocated to pods that used them.

## Solution
- **Detect PV-only assets:**  
  Skip idle allocation for assets where  
  `PersistentVolumeCost > 0 && AttachedVolumeCost == 0`
- **Skip idle processing for PVs:**  
  PV-only assets don't need node-level idle processing since costs are already allocated to pods.
- **Add debug logging:**  
  Aid in troubleshooting any remaining edge cases.
- **Preserve existing behavior:**  
  Attached volumes still follow normal node-level processing.

### Test Setup
```bash
# Created test PVC and pod
kubectl get pvc test-pvc-allocation-fix
# NAME                      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS
# test-pvc-allocation-fix   Bound    pvc-4d716511-7cd7-4529-8ec6-75d48f460b23   1Gi        RWO            standard

# Test PVC UID: 4d716511-7cd7-4529-8ec6-75d48f460b23
````

### Validation Results

```bash
# ✅ Before fix (expected warnings):
WRN ETL: did not find allocations for asset key: default/pvc-4d716511-7cd7-4529-8ec6-75d48f460b23

# ✅ After fix (20+ minutes runtime, ZERO PVC warnings):
grep "did not find allocations.*pvc-" <opencost_logs>
# (no output - fix working!)

# ✅ API confirms correctness:
curl -s "http://localhost:9003/assets?window=1h" | jq '.data | length'
# 4

curl -s "http://localhost:9003/allocation?window=1h" | jq '.data | length'
# 1
```

## Impact

* ✅ Tested locally with bound PVCs for 20+ minutes.
* ✅ Zero false warnings generated.
* ✅ PVC costs remain correctly allocated to pods.
* ✅ Attached volume processing unchanged.
* ✅ All existing functionality preserved.

**Result:**
Eliminates false warnings for all bound PVCs, improves log cleanliness, and reduces user confusion — without changing cost allocation accuracy.

Fixes https://github.com/opencost/opencost/issues/3211

---

> [!NOTE]
> Prevents false warnings and idle processing for PV-only assets during idle allocation calculation.
> 
> - In `computeIdleAllocations`, skip idle allocation when `PersistentVolumeCost > 0 && AttachedVolumeCost == 0`, logging a debug message instead of a warning
> - Add debug log to list available allocation keys when a mismatch occurs
> - New tests: `TestComputeIdleAllocations_PVCSkipLogic` and `TestPVCAllocationSkipLogic` verify the skip condition and behavior
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 62fb0dab0f440e4849655206879def24ad38df61. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by fromsaurav** [comment]
@ameijer , Could you help with the CI workflow issues? The `set-follow-up-label` action seems to have permission problems and not related to this pr,  my local testing shows the fix working correctly.

File: pkg/costmodel/costmodel.go [hunk 1]
```diff
@@ -1757,20 +1757,35 @@ func computeIdleAllocations(allocSet *opencost.AllocationSet, assetSet *opencost
...
+
+				// Use a zero-value set of totals. This indicates either (1) an
+				// error computing totals, or (2) that no allocations ran on the
+				// given node for the given window.
+				allocTotal = &opencost.AllocationTotals{
+					Cluster: assetTotal.Cluster,
+					Node:    assetTotal.Node,
+					Start:   assetTotal.Start,
+					End:     assetTotal.End,
+				}
```
Comment (ameijer) [inline pkg/costmodel/costmodel.go]: @fromsaurav would you be able to unit test this? I think you could cook the inputs pretty easily...

Comment (fromsaurav) [reply pkg/costmodel/costmodel.go]: > @fromsaurav would you be able to unit test this? I think you could cook the inputs pretty easily...

Would i do the unit tests in a separate commit on this PR, or as a follow-up PR? I can implement either,

Comment (ameijer) [reply pkg/costmodel/costmodel.go]: please add them to this PR so we can see the effect of your changes in context!

Comment (fromsaurav) [reply pkg/costmodel/costmodel.go:1787]: sure, thank you!

Comment (fromsaurav) [reply pkg/costmodel/costmodel.go]: > please add them to this PR so we can see the effect of your changes in context!

sure, thank you!

Comment (fromsaurav) [reply pkg/costmodel/costmodel.go]: @ameijer added unit tests 👍

**Comment by lmello** [comment]
thanks for fixing this.

**Review by ameijer — APPROVED** [review]
LGTM

**Comment by ameijer** [comment]
@fromsaurav looks like some build failures?

**Comment by ameijer** [comment]
@fromsaurav looks like you have test failures here
